test/unit/messaging/messaging.spec.ts

```diff
@@ -485,19 +487,25 @@ describe('Messaging', () => {
     });
 
     it('should fail when the backend server returns a detailed error', () => {
-      const resp = {
-        error: {
-          status: 'INVALID_ARGUMENT',
-          message: 'test error message',
-        },
-      };
-      mockedRequests.push(mockSendError(400, 'json', resp));
+      generatedClientStub = sinon
+        .stub(FcmServiceClient.prototype, 'sendMessage')
+        .rejects({
+          'message': 'The registration token is not a valid FCM registration token',
+          'code': 400,
+          'status':'INVALID_ARGUMENT',
+          'details': [
+            { '@type':'type.googleapis.com/google.firebase.fcm.v1.FcmError',
+              'errorCode':'INVALID_ARGUMENT'
+            }
+          ],
+          'note': 'Exception occurred in retry method that was not classified as transient'
+        });
       return messaging.send(
         { token: 'mock-token' },
-      ).should.eventually.be.rejectedWith('test error message')
-        .and.have.property('code', 'messaging/invalid-argument');
+      ).should.eventually.be.rejectedWith('The registration token is not ' +
```

Comment:
Should also assert for the error code here.

Reply:
Added

Reply:
Added